ang/civicase/Utils.js

```diff
@@ -345,19 +370,29 @@
           'activity_type_id.grouping': {LIKE: '%task%'},
           'status_id.filter': 0,
           options: {limit: panelLimit, sort: 'activity_date_time ASC'},
-          return: ['activity_type_id', 'subject', 'activity_date_time', 'status_id', 'target_contact_name', 'assignee_contact_name', 'is_overdue', 'is_star', 'file_id', 'tag_id.name', 'tag_id.description', 'tag_id.color', 'file_id']
+          return: activityReturnParams
+        },
+        // For the "Next Activity" panel
+        'api.Activity.get.4': {
+          case_id: '$value.id',
+          status_id: {'!=': 'Completed'},
+          options: {
```

Comment:
Why dont you send a filter param so that it only returns non milestone values. that way you can just limit the value to `1`. Instead of getting all.
Some thing like `'activity_type_id.grouping': {NOT LIKE: '%milestone%'},`. That way we dont have to do anything on the frontend to filter.